ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/rlpx/connections/netty/DeFramer.java

```diff
@@ -129,13 +134,30 @@ protected void decode(final ChannelHandlerContext ctx, final ByteBuf in, final L
                 subProtocols,
                 localNode.getPeerInfo().getCapabilities(),
                 peerInfo.getCapabilities());
-        final Optional<Peer> peer = expectedPeer.or(() -> createPeer(peerInfo, ctx));
-        if (peer.isEmpty()) {
-          LOG.debug("Failed to create connection for peer {}", peerInfo);
-          connectFuture.completeExceptionally(new PeerChannelClosedException(peerInfo));
-          ctx.close();
-          return;
+
+        Optional<Peer> peer;
+        if (expectedPeer.isPresent()) {
+          peer = expectedPeer;
+
+        } else {
+          // If the expected peer is not set the connection was inbound initiated.
+          // There is a good chance that we have bonded with that peer, so
+          // try to find the peer in the PeerTable, as it might contain additional
+          // information, like the fork id.
```

Comment:
this comment doesn't quite match the code. looks like we're only checking the peerTable if createPeer() doesn't work?

Reply:
Changed the comment. Please have a look if that is better!